server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java

```diff
@@ -19,20 +19,97 @@
+import org.apache.james.jmap.model.MailboxCreationId;
+import org.apache.james.jmap.model.SetError;
 import org.apache.james.jmap.model.SetMailboxesRequest;
 import org.apache.james.jmap.model.SetMailboxesResponse;
 import org.apache.james.jmap.model.mailbox.Mailbox;
+import org.apache.james.jmap.model.mailbox.MailboxRequest;
+import org.apache.james.jmap.utils.MailboxHierarchySorter;
+import org.apache.james.jmap.utils.MailboxUtils;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxPath;
 import org.apache.james.mailbox.store.mail.model.MailboxId;
 
+import com.github.fge.lambdas.Throwing;
+import com.google.common.annotations.VisibleForTesting;
+
 public class SetMailboxesCreationProcessor<Id extends MailboxId> implements SetMailboxesProcessor<Id> {
 
-    public SetMailboxesResponse process(SetMailboxesRequest request) {
+    private final MailboxManager mailboxManager;
+    private final MailboxHierarchySorter<Map.Entry<MailboxCreationId, MailboxRequest>, String> mailboxHierarchySorter;
+    private final MailboxUtils<Id> mailboxUtils;
+
+    @Inject
+    @VisibleForTesting
+    SetMailboxesCreationProcessor(MailboxManager mailboxManager, MailboxUtils<Id> mailboxUtils) {
+        this.mailboxManager = mailboxManager;
+        this.mailboxHierarchySorter =
+            new MailboxHierarchySorter<>(
+                x -> x.getKey().getCreationId(),
+                x -> x.getValue().getParentId());
+        this.mailboxUtils = mailboxUtils;
+    }
+
+    public SetMailboxesResponse process(SetMailboxesRequest request, MailboxSession mailboxSession) {
         SetMailboxesResponse.Builder builder = SetMailboxesResponse.builder();
-        request.getCreate().entrySet().stream()
-            .forEach(entry -> builder.creation(
-                entry.getKey(),
-                Mailbox.builder().name(entry.getValue().getName()).id("serverId").build()));
+        Map<MailboxCreationId, String> creationIdsToCreatedMailboxId = new HashMap<>();
+        mailboxHierarchySorter.sortFromRootToLeaf(request.getCreate().entrySet())
+            .forEach(entry -> {
+                createMailbox(entry.getKey(), entry.getValue(), mailboxSession, creationIdsToCreatedMailboxId, builder);
+            });
         return builder.build();
     }
 
+    private void createMailbox(MailboxCreationId mailboxCreationId, MailboxRequest mailboxRequest, MailboxSession mailboxSession,
+            Map<MailboxCreationId, String> creationIdsToCreatedMailboxId, SetMailboxesResponse.Builder builder) {
+        try {
+            MailboxPath mailboxPath = getMailboxPath(mailboxRequest, creationIdsToCreatedMailboxId, mailboxSession);
+            mailboxManager.createMailbox(mailboxPath, mailboxSession);
+            Optional<Mailbox> mailbox = mailboxUtils.mailboxFromMailboxPath(mailboxPath, mailboxSession);
```

Comment:
Used in both way: by lambda or not.
I don't see the advantage of such modifications.